### PR TITLE
docs(DTO): Replace PartialDataDTO with DTOData

### PR DESCRIPTION
**Pull request by ddxv**
…itestar.dev/latest/tutorials/dto-tutorial/09-updating.html\#patch-handlers

### Pull Request Checklist

[//]: # "By submitting this issue, you agree to:"
[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

### Description
[//]: # "Please describe your pull request for new release changelog purposes"
[//]: # "Example: 'This pr adds the capability to use server-sent events, and documents it in the usage docs'"

This PR fixes an apparent typo in the Extended Example on the index.rst homepage.

When running the hello world Extended Example, I was unable to proceed due to this error:
```litestar.exceptions.dto_exceptions.InvalidAnnotationException: DTO narrowed with '<class 'my_app.models.User'>', handler type is '<class 'my_app.models.PartialUserDTO'>'```

Discord Channel User Alc pointed me in the direction of a fix based on:
[https://litestar.dev/latest/tutorials/dto-tutorial/09-updating.html\#patch-handlers](https://docs.litestar.dev/latest/tutorials/dto-tutorial/09-updating.html#patch-handlers)

-

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

-

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -191,7 +191,7 @@ You can also use dataclasses (standard library and Pydantic),
             ...
 
         @patch(path="/{user_id:uuid}", dto=PartialUserDTO)
-        async def partial_update_user(self, user_id: UUID4, data: PartialUserDTO) -> User:
+        async def partial_update_user(self, user_id: UUID4, data: DTOData) -> User:
```
Comment (Alc-Alc) [inline docs/index.rst:194]: Would be better if the import for this was also added? `from litestar.dto import DTOData`?

**Review by Alc-Alc — COMMENTED** [review]
Thanks, just a minor nitpick with the import statement 🙂

File: docs/index.rst (hunk 1 above)
Comment (ddxv) [reply docs/index.rst]: -Also DTOData[User]?
Ah thanks, I see that is a mistake by me, I will fix.

-Would be better if the import for this was also added?
There is already a `from litestar.dto import DTOData` on line 176, does this work in this context?

**Review by ddxv — COMMENTED** [review]
Fixed DataDTO to DataDTO[User]

File: docs/index.rst (hunk 1 above)
Comment (Alc-Alc) [reply docs/index.rst]: ahh ok, I must have missed that, thanks again

**Review by provinzkraut — APPROVED** [review]

**Pull request merged**